core/network/src/protocol.rs

```diff
@@ -14,36 +14,37 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-use std::collections::{HashMap, HashSet, BTreeMap};
-use std::cmp;
-use std::io::Cursor;
-use std::sync::Arc;
-use std::time;
-use parking_lot::RwLock;
-use rustc_hex::ToHex;
-use runtime_primitives::traits::{Block as BlockT, Header as HeaderT, NumberFor, As, Zero};
-use runtime_primitives::generic::BlockId;
-use primitives::storage::StorageKey;
+use codec::Encode;
+use crossbeam_channel::{tick, unbounded, Receiver, Sender};
```

Comment:
style: I would prefer to `use crossbeam_channel::{self as channel, Receiver, Sender}` and use `channel::unbounded()` and `channel::tick()` to make context more clear.